### PR TITLE
Refactor stale statuses

### PR DESCRIPTION
**Pull request by PaulFarault**
#### Which issue(s) this PR fixes

Fixes None

#### Additional comments

Remove `ClusterStatus` parameter from the `Deployment.from_stale_components` method.

#### Agreements

- [X] I hereby declare this contribution to be licensed under the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0).
- [X] I hereby agree to grant [TOSIT](https://www.tosit.io/) a copyright license to use my contributions.

### CODE REVIEW

**Comment by SteBaum** [comment]
I have `Error: running_version` when I modified the variables of spark3 and tried to do a `tdp plan reconfigure` or also a `tdp plan ops spark3_client_install --host=sbaum-edge-01` which works but the `tdp deploy --mock-deploy` fails afterwards while it works on the master branch.

**Comment by PaulFarault** [comment]
Thanks! This was due to an error when extracting attributes from the query result. Type inferring with SQL Alchemy is not always working unfortunately. I fixed it, everything should work as expected from now !

**Review by SteBaum — APPROVED** [review]
Reviewed and tested

File: tdp/core/models/deployment_model.py [hunk 1]
```diff
@@ -327,49 +327,30 @@ def from_operations_hosts_vars(
         return deployment
 
     @staticmethod
-    def from_stale_components(
+    def from_stale_hosted_entitites(
```
Comment (rpignolet) [inline tdp/core/models/deployment_model.py:330]: Typo
```suggestion
    def from_stale_hosted_entities(
```

Comment (PaulFarault) [reply tdp/core/models/deployment_model.py]: Fixed. Thx!

File: tdp/core/entities/hostable_entity_name.py [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+class HostableEntityName(ABC):
+    @property
+    @abstractmethod
+    def name(self) -> str:
+        pass
+
+
+@dataclass(frozen=True)
+class ServiceName(HostableEntityName):
+    _name: str
```
Comment (rpignolet) [inline tdp/core/entities/hostable_entity_name.py:22]: Why there is a `_name` here. `ComponentName` use `name`.

**Pull request merged**